### PR TITLE
test: Removes several multiprocessing executions of test_enqueuer.

### PR DESCRIPTION
**Pull request by wes-turner**
On a mac, the tests where `multiprocessing=True` run slow:

```
use_multiprocessing=False: 72 passed ... in 2.22s
use_multiprocessing=True: 72 passed ... in 144.54s
```

Since there's not likely to be much interaction between this parameter and the others, this patch eliminates 71 of the
`use_multiprocessing=True` test cases. `coverage` shows no decrease in code coverage.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

MLG-578

### CODE REVIEW

File: harness/tests/experiment/keras/test_keras_data.py [hunk 1]
```diff
@@ -174,7 +173,41 @@ def test_enqueuer(
...
+def test_enqueuer_multiprocessing() -> None:
+    """Same enqueuer test as above, but with multiprocessing enabled.
+
+    A single, arbitrary set of parameters from the above test is chosen in order to exercise the
+    keras.enqueuer._MultiprocessingEnqueuer. In this test, multiprocessing is slow on macOS, so
+    we're minimizing the number of times we execute it assuming there's no interaction between the
+    choice of threading/multiprocessing and the examples loaded by the enqueuer.
+    """
+    epoch_len = 100
+    workers = 5
```
Comment (MikhailKardash) [inline harness/tests/experiment/keras/test_keras_data.py:199]: I'm not sure if having just 1 multiprocessing passthrough is enough to fully test the enqueuer functionality.
Looking at the actual class, it seems to me like `shuffle` , `skip` , and `rank size` are properties of the sampler rather than the actual enqueuer, so I have no problems with keeping those fixed for a multiprocessing test of the enqueuer. 
However, I do think we should at least sweep the `workers` parameter.

Comment (MikhailKardash) [reply harness/tests/experiment/keras/test_keras_data.py:199]: > it's also kind of more attention than I want to give this problem.
I agree.
> I don't really want to sweep workers for use_multiprocessing=True because of this conditional
Ah, I was so focused on the sampler that I didn't notice that one. Then just nix the workers=0 case since that is also already tested by the threading test.

File: harness/tests/experiment/keras/test_keras_data.py [hunk 2]
```diff
@@ -157,13 +157,12 @@ def test_sampler(shuffle: bool, skip: int, rank_size: Tuple[int, int]) -> None:
...
 
 
-@pytest.mark.parametrize("use_multiprocessing", [False, True])
 @pytest.mark.parametrize("workers", [0, 1, 5])
 @pytest.mark.parametrize("rank_size", [(0, 1), (0, 3), (1, 3), (2, 3)])
 @pytest.mark.parametrize("skip", [0, 50, 350])
 @pytest.mark.parametrize("shuffle", [False, True])
-def test_enqueuer(
-    shuffle: bool, skip: int, rank_size: Tuple[int, int], workers: int, use_multiprocessing: bool
+def test_enqueuer_threading(
```
Comment (MikhailKardash) [inline harness/tests/experiment/keras/test_keras_data.py:164]: I believe `test_build_enqueuer` is probably more appropriate here given the parameter sweep.

File: harness/tests/experiment/keras/test_keras_data.py (hunk 1 above)
Comment (wes-turner) [reply harness/tests/experiment/keras/test_keras_data.py]: done

File: harness/tests/experiment/keras/test_keras_data.py [hunk 3]
```diff
@@ -174,7 +171,41 @@ def test_enqueuer(
...
+        assert list(enqueuer.data()) == list(sampler.yield_epoch()), "first epoch was wrong"
+        assert list(enqueuer.data()) == list(sampler.yield_epoch()), "second epoch was wrong"
+        assert list(enqueuer.data()) == list(sampler.yield_epoch()), "third epoch was wrong"
+
+
+@pytest.mark.parametrize("workers", [1, 5])
+def test_enqueuer_multiprocessing(workers: int) -> None:
+    """Same enqueuer test as above, but with multiprocessing enabled.
+
+    A single, arbitrary set of parameters from the above test is chosen in order to exercise the
```
Comment (MikhailKardash) [inline harness/tests/experiment/keras/test_keras_data.py:192]: This docstring needs a slight update, but otherwise LGTM

**Review by MikhailKardash — APPROVED** [review]

File: harness/tests/experiment/keras/test_keras_data.py (hunk 3 above)
Comment (wes-turner) [reply harness/tests/experiment/keras/test_keras_data.py:192]: Ooh. Thanks for catching that.
